python/setup.py

```diff
@@ -0,0 +1,29 @@
+from setuptools import setup, find_packages
+from setuptools.dist import Distribution
+
+class BinaryDistribution(Distribution):
+    """Distribution which always forces a binary package with platform name
+    """
+    def has_ext_modules(self):
+        return True
+
+path_to_file = "../pointmatcher/PointMatcher.h"
+
+version = None
+
+with open(path_to_file, "r", encoding="utf-8") as f:
```

Comment:
Fetching the pointmatcher version dynamicaly will be a very usefull tool for the build system automation.
(!) Limitation: This approach to fetching the `POINTMATCHER_VERSION` only work if the user execute

``` bash
python -m build --wheel
```

It will break when using the default

``` bash
python -m build
```

because the build command will build *sdist* in a first step and then build *wheel* fetching from a temporary dir `/tmp/build-via-sdist-*/pypointmatcher-*`.
I recommand
- to keep this approach for now but wrapped in a try/except clause with an explicit error message pinpointing its origin since it is not obvious when it happen in a install stacktrace which is usualy prety long;
- to add an explicit comment on those limitations in the code to advise futur developer;
- and to add a warning in the `doc/CompilationPython.md` package install procedure.